daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -305,6 +305,22 @@ private[lf] object SBuiltin {
     }
   }
 
+  final case object SBToTextContractId extends SBuiltin(1) {
+    override private[speedy] final def execute(
+        args: util.ArrayList[SValue],
+        machine: Machine): Unit = {
+      args.get(0) match {
+        case SContractId(cid) =>
+          if (machine.onLedger) {
+            machine.returnValue = SOptional(None)
+          } else {
+            machine.returnValue = SOptional(Some(SText(cid.coid)))
+          }
```

Comment:
```suggestion
      machine.returnValue = args.get(0) match {
        case SContractId(cid) if (machine.onLedger) =>
             SOptional(Some(SText(cid.coid)))
        case _ => 
             SValue.SValue.None
      }  
```

Reply:
This suggestion isn’t correct.
1. You flipped the flag. You are returning `Some` in the `onLedger` case.
2. I would like to validate that we get a contract id even in the onLedger case. Just makes it easier to catch bugs.

Reply:
My bad.
You could still use `SValue.SValue.None` that save you an allocation.

Reply:
Thanks, changed!